core/aws_ddk_core/cicd/pipeline.py

```diff
@@ -335,6 +342,43 @@ def add_checks(self) -> "CICDPipelineStack":
             self.add_test_stage()
         return self
 
+    def add_custom_stage(self, stage_name: str, steps: List[Dict[str, Any]]) -> "CICDPipelineStack":
```

Comment:
I'm hoping this was the right way to do the types. It did not work when I was explicit with more than two attributes i.e. `List[Dict[str, List[str], Optional[IFileSetProducer], Optional[List[str]]]]`